### PR TITLE
Support for mqtt publisher in network scripts

### PR DESCRIPTION
**Pull request by pfl**
Support the non-TLS version of mqtt_publisher in the network testing script. The latest version of net-tools is required in west.yaml, as that one contains the mosquitto MQTT server. The README has also been updated with instructions on how to create the net-tools Docker image.

### CODE REVIEW

**Comment by jukkar** [comment]
fyi @rlubos

File: samples/net/mqtt_publisher/Kconfig [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+# SPDX-License-Identifier: Apache-2.0
+
+mainmenu "Networking mqtt_publisher sample application"
+
+config APP_MAX_ITERATIONS
+        int "Number of times to Publish sample MQTT messages"
+        default 500
+        help
+          Send sample MQTT messages this many times in a MQTT connection
+	  before exiting. A value of zero means that the messages are sent
```
[the remark this answers is not in the document]
Comment (pfl) [reply samples/net/mqtt_publisher/Kconfig]: Tabs, missing.

File: samples/net/mqtt_publisher/Kconfig [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+	  forever.
+
+config APP_MAX_CONNECTIONS
+       int "Number of times to connect to the MQTT server"
+       default 0
+       help
+         Number of times to connect to the MQTT server. With each connection
+	 send APP_MAX_ITERATIONS amount of MQTT sample messages. A value of
```
[the remark this answers is not in the document]
Comment (pfl) [reply samples/net/mqtt_publisher/Kconfig]: Same here.

**Comment by pfl** [comment]
> +	exit (r);

Ah, that one too.

**Review by jukkar — APPROVED** [review]

File: samples/net/mqtt_publisher/Kconfig [hunk 3]
```diff
@@ -0,0 +1,24 @@
+# Config options for mqtt_publisher sample application
+
+# Copyright (c) 2020 Intel Corporation
+# SPDX-License-Identifier: Apache-2.0
+
+mainmenu "Networking mqtt_publisher sample application"
+
+config APP_MAX_ITERATIONS
```
Comment (jukkar) [inline samples/net/mqtt_publisher/Kconfig]: Missed this one. We have actually called these sample app variables like this `NET_SAMPLE_...` in other samples so we could follow the same style.

**Review by jukkar — CHANGES_REQUESTED** [review]

File: samples/net/mqtt_publisher/Kconfig (hunk 3 above)
Comment (pfl) [reply samples/net/mqtt_publisher/Kconfig]: Fixed.

**Review by jukkar — APPROVED** [review]
LGTM

**Review by tbursztyka — APPROVED** [review]

**Comment by jukkar** [comment]
ping @nashif as a codeowner of files under scripts/, are you ok with these changes?

**Pull request merged**